samples/Common.c

```diff
@@ -34,18 +37,123 @@ STATUS signalingCallFailed(STATUS status)
 
 VOID onDataChannelMessage(UINT64 customData, PRtcDataChannel pDataChannel, BOOL isBinary, PBYTE pMessage, UINT32 pMessageLen)
 {
-    UNUSED_PARAM(customData);
-    if (isBinary) {
-        DLOGI("DataChannel Binary Message");
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT32 i, strLen, tokenCount;
+    CHAR pMessageSend[SIZEOF(DataChannelMessage)];
+    PCHAR json;
+    PSampleStreamingSession pSampleStreamingSession = (PSampleStreamingSession) customData;
+    PSampleConfiguration pSampleConfiguration = pSampleStreamingSession->pSampleConfiguration;
+    DataChannelMessage dataChannelMessage;
```

Comment:
We should have a NULL check here. If NULL, we should send the other end a message saying `unable to send metrics` or something like that.

Reply:
Added a NULL check!